core/src/main/scala/org/typelevel/paiges/Chunk.scala

```diff
@@ -7,28 +7,34 @@ private[paiges] object Chunk {
   /**
    * Given a width and Doc find the Iterator
    * of Chunks.
+   *
+   * A `trim` document is one where all lines consisting entirely of indentation are
+   * represented by the empty string.  Note that this does *not* in general include lines consisting
+   * solely of whitespace.  E.g. Doc.text(" ") will always insert a space, regardless
+   * of whether the document is trimmed or not.
    */
-  def best(w: Int, d: Doc): Iterator[String] = {
+  def best(w: Int, d: Doc, trim: Boolean): Iterator[String] = {
 
     val nonNegW = w max 0
 
     sealed abstract class ChunkStream
     object ChunkStream {
       case object Empty extends ChunkStream
-      case class Item(str: String, position: Int, stack: List[(Int, Doc)], isBreak: Boolean) extends ChunkStream {
+      case class Item(str: String, position: Int, cache: Option[ChunkStream], stack: List[(Int, Doc)], isBreak: Boolean) extends ChunkStream {
```

Comment:
Done.  (Is using null just a fact of life when trying to write fast Scala code?  So sad...)